libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Security/SFSecureEncryptionKey.m

```diff
@@ -0,0 +1,193 @@
+/*
+ Copyright (c) 2019-present, salesforce.com, inc. All rights reserved.
+ 
+ Redistribution and use of this software in source and binary forms, with or without modification,
+ are permitted provided that the following conditions are met:
+ * Redistributions of source code must retain the above copyright notice, this list of conditions
+ and the following disclaimer.
+ * Redistributions in binary form must reproduce the above copyright notice, this list of
+ conditions and the following disclaimer in the documentation and/or other materials provided
+ with the distribution.
+ * Neither the name of salesforce.com, inc. nor the names of its contributors may be used to
+ endorse or promote products derived from this software without specific prior written
+ permission of salesforce.com, inc.
+ 
+ THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR
+ IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND
+ FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+ CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+ WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY
+ WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#import "SFSecureEncryptionKey.h"
+#import "SFSDKCryptoUtils.h"
+#import "NSData+SFAdditions.h"
+
+// NSCoding constants
+static NSString * const kSecureEncryptionKeyCodingValue = @"com.salesforce.encryption.securekey.label";
+
+@interface SFSecureEncryptionKey () {
+    SecKeyRef publicKeyRef;
+    SecKeyRef privateKeyRef;
+}
+
+@property (nonatomic, strong) NSString *label;
+
+@end
+
+@implementation SFSecureEncryptionKey
+
+#pragma mark - Factory methods and constructor
+
++ (instancetype) createKey:(NSString*)label
+{
+    [SFSDKCryptoUtils deleteECKeyPairWithName:label]; // Delete existing key if any
```

Comment:
Should this delete and recreate if it already exists, or just leave it alone and return the existing instance, i.e. call `retrieveKey` instead?

Reply:
We can go either way. I don't have a preference.

Reply:
I would probably go with leaving it alone and returning the existing instance, but I'll leave it up to you.